lib/linter-views.coffee

```diff
@@ -94,22 +93,23 @@ class LinterViews
   updateTabs: ->
     first = null
     last = null
-    foundActive = false
-    firstLabel = null
-    for key, tab of @tabs # for...of (key, value)
+
+    for key, tab of @tabs
       tab.classList.remove('first')
       tab.classList.remove('last')
       tab.visibility = atom.config.get("linter.showErrorTab#{key}")
       if tab.visibility
         last = tab
-        foundActive = foundActive || tab.active
-        unless first
-          first = tab
-          firstLabel = key
-    @changeTab(firstLabel) if first and not foundActive
+        first = tab unless first
+
     first.classList.add('first') if first
     last.classList.add('last') if last
 
+    unless @tabs[@state.scope]?.visibility
+      @state.scope = @tabPriority.filter((key) => @tabs[key].visibility)[0]
+
+    @changeTab(@state.scope, false)
```

Comment:
How does it handle if there's no active tab btw?
Does it still show the results from the last active tab?